osu.Game/Screens/Multi/Play/TimeshiftPlayer.cs

```diff
@@ -95,19 +96,38 @@ protected override ResultsScreen CreateResults(ScoreInfo score)
             return new TimeshiftResultsScreen(score, roomId.Value.Value, playlistItem, true);
         }
 
-        protected override ScoreInfo CreateScore()
+        protected override Score CreateScore()
         {
             var score = base.CreateScore();
-            score.TotalScore = (int)Math.Round(ScoreProcessor.GetStandardisedScore());
+            score.ScoreInfo.TotalScore = (int)Math.Round(ScoreProcessor.GetStandardisedScore());
+            return score;
+        }
 
+        protected override async Task<ScoreInfo> SubmitScore(Score score)
+        {
             Debug.Assert(token != null);
 
-            var request = new SubmitRoomScoreRequest(token.Value, roomId.Value ?? 0, playlistItem.ID, score);
-            request.Success += s => score.OnlineScoreID = s.ID;
-            request.Failure += e => Logger.Error(e, "Failed to submit score");
+            bool completed = false;
+            var request = new SubmitRoomScoreRequest(token.Value, roomId.Value ?? 0, playlistItem.ID, score.ScoreInfo);
+
+            request.Success += s =>
+            {
+                score.ScoreInfo.OnlineScoreID = s.ID;
+                completed = true;
+            };
+
+            request.Failure += e =>
+            {
+                Logger.Error(e, "Failed to submit score");
+                completed = true;
+            };
+
             api.Queue(request);
 
-            return score;
+            while (!completed)
```

Comment:
is there a reason for using `Task.Delay` rather than a manual reset event?

Reply:
I don't want to be completely blocking the threadpool thread via an MRE, but I've used a TaskCompletionSource instead since it's awaitable.